packages/block-editor/src/components/block-edit/edit.js

```diff
@@ -48,27 +57,225 @@ const Edit = ( props ) => {
 const EditWithFilters = withFilters( 'editor.BlockEdit' )( Edit );
 
 const EditWithGeneratedProps = ( props ) => {
-	const { attributes = {}, name } = props;
+	const { name, clientId, attributes, setAttributes } = props;
+	const registry = useRegistry();
 	const blockType = getBlockType( name );
 	const blockContext = useContext( BlockContext );
+	const registeredSources = useSelect( ( select ) =>
```

Comment:
The only slightly concerning thing is the extra blocks store sub. But this store usually almost never changes, so it's not as bad as a blockEditor store sub. I don't think it will affect the results.

Reply:
Right, it's a trade-off.